__tests__/db.test.js

```diff
@@ -52,8 +106,219 @@ describe("GET /api", () =>{
+            expect(new Date(body.articles[3].created_at).getTime() >= new Date(body.articles[4].created_at).getTime()).toBe(true)
+            expect(new Date(body.articles[4].created_at).getTime() >= new Date(body.articles[6].created_at).getTime()).toBe(true)
+        })
+    })
+    test("Bad Queries do not effect outcome length or status", () =>{
+        return request(app)
+        .get('/api/articles?article_id=ga')
+        .then((res)=>{
+                expect(res.statusCode).toBe(200)
+                expect(res.body.articles.length > 0)
+        })
+    })
+  })
+describe("GET /api/articles/", () => {
+    test("Calls appropriate function with 200 status code", () =>{
+        return request(app)
+        .get('/api/articles/1')
+        .then((res)=>{
+            expect(res.statusCode).toBe(200)
+        })
+    })
+    test("Returns an object in the body", () =>{
+        return request(app)
+        .get('/api/articles/1')
+        .then(({body})=>{
+            expect(typeof body).toBe("object")
+        })
+    })
+    test("Returns 400 status code when requesting a non-numeric ID", () =>{
+        return request(app)
+        .get('/api/articles/fish')
+        .then((res)=>{
+            expect(res.statusCode).toBe(400)
+        })
+    })
+    test("Returns 404 status code when requesting an article ID with no data", () =>{
+        return request(app)
+        .get('/api/articles/135477')
+        .then((res)=>{
+            expect(res.statusCode).toBe(404)
+        })
+    })
+    test("Returns an article with the correct properties", () =>{
+        return request(app)
+        .get('/api/articles/1')
+        .then(({body})=>{
+            expect(body.article).toHaveProperty('author')
+            expect(body.article).toHaveProperty('title')
+            expect(body.article).toHaveProperty('article_id')
+            expect(body.article).toHaveProperty('body')
+            expect(body.article).toHaveProperty('topic')
+            expect(body.article).toHaveProperty('created_at')
+            expect(body.article).toHaveProperty('votes')
+            expect(body.article).toHaveProperty('article_img_url')
+        })
+    })
+})
 
+describe("POST /api/articles/:article_id/comments", () =>{
+    test("Returns a 201 when posting data", ()=>{
+        return request(app).post("/api/articles/1/comments").send({username:"lurker", body:"Sorry -snip-"})
+            .then((res)=>{
+                expect(res.statusCode).toBe(201)
+        })
+    })
+    test("Returns only the comment when successfully posted", ()=>{
+        return request(app).post("/api/articles/1/comments").send({username:"lurker", body:"Sorry -snip-"})
+            .then((res)=>{
+                expect(res.statusCode).toBe(201)
+                expect(res.body.postedComment).toBe("Sorry -snip-")
+                expect(Object.keys(res.body)).toEqual(["postedComment"])
+        })
```

Comment:
Realistically, you want to send back the entire comment, including username, body, comment_id, etc - everything that has now been successfully gathered from the db